inspirehep/base/templates/format/record/Inspire_HTML_detailed_macros.tpl

```diff
@@ -222,30 +220,59 @@
 {% macro record_links() %}
   {% if record.get('urls') %}
     {% for url in record.get('urls') %}
-      {% if ( (loop.index == 1) and ( (url.get('url').startswith('http://www.adsabs.') or url.get('url').startswith('http://www-public.slac.stanford.edu') ) ) )  %}
-        View in
-      {% endif %}
-      {% if url.get('url').startswith('http://www.adsabs.') %}
-        <a href='{{ url.get('url') }}'>ADS</a>
-      {% endif %}
-      {% if url.get('url').startswith('http://www-public.slac.stanford.edu') %}
-        <a href='{{ url.get('url') }}'>SLAC</a>
-      {% endif %}
-      {% if url.get('url').startswith('http://www.ams.org') %}
-        <a href='{{ url.get('url') }}'>AMS MathSciNet</a>
-      {% endif %}
-      {% if url.get('url').startswith('http://www.zentralblatt-math.org') %}
-        <a href='{{ url.get('url') }}'>zbMATH</a>
-      {% endif %}
-      {% if url.get('url').startswith('http://projecteuclid.org') %}
-        <a href='{{ url.get('url') }}'>Project Euclid</a>
-      {% endif %}
-      {% if url.get('url').startswith('http://www-lib.kek.jp') %}
-        <a href='{{ url.get('url') }}'>KEK scanned document</a>
-      {% endif %}
-      {% if url.get('url').startswith('http://cds.cern.ch/record/') %}
-        <a href='{{ url.get('url') }}'>CDS</a>
-      {% endif %}
+    {% if url.get('url') != None %}
+     {% if url.get('url')|is_list() %}
```

Comment:
As with the `dois`, can you have a look at why that field would ever not be a list?

In `dojson` we have:

```
@utils.for_each_value
```

So it should always be a list.

Solving the source of the problem would avoid so much duplication in the jinja templates.